### PR TITLE
VAULT-5038 Convert to CRT for releases

### PR DESCRIPTION
**Pull request by swenson**
# Overview
Convert to CRT for releases, as mandated.

* We remove CircleCI support, since it will no longer be used.
* Add in GitHub Actions for testing and building an artifact.
* Upgrade Go from 1.17.2 to 1.17.7 for consistency with Vault.

There should be no user-facing changes.

# Design of Change
We changed some of the build scripts and the `Dockerfile` to more closely conform to the CRT requirements. A few more files can probably be deleted (e.g., `scripts/dist.sh`) after merging this once we confirm we don't need them anymore.

# Contributor Checklist
[x] Add relevant docs to upstream Vault repository, or sufficient reasoning why docs won’t be added yet (not needed)
[x] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
[x] Backwards compatible

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -64,20 +64,20 @@ if [ "${VAULT_DEV_BUILD}x" != "x" ]; then
...
-    echo "==> Packaging..."
-    for PLATFORM in $(find ./pkg/bin -mindepth 1 -maxdepth 1 -type d); do
-        OSARCH=$(basename ${PLATFORM})
-        echo "--> ${OSARCH}"
+# if [ "${VAULT_DEV_BUILD}x" = "x" ]; then
+#     # Zip and copy to the dist dir
+#     echo "==> Packaging..."
+#     for PLATFORM in $(find ./pkg/bin -mindepth 1 -maxdepth 1 -type d); do
+#         OSARCH=$(basename ${PLATFORM})
+#         echo "--> ${OSARCH}"
```
[the remark this answers is not in the document]
Comment (swenson) [reply scripts/build.sh:72]: This creates a ZIP, which is created elsewhere now. I can delete this block now.

**Comment by swenson** [comment]
I'm still waiting on access to the staging AWS account so I can do the promotion.

I'll remove the Docker parts though once I can verity that we don't need them.

**Comment by swenson** [comment]
Okay, I confirmed this workflow succeeded at publishing to staging:

https://artifactory.hashicorp.engineering/ui/native/hashicorp-crt-staging-local/vault-plugin-database-oracle/0.0.0-dev/bbea3baf828a239fa2eed77fab8afead933672cc/

**Review by tomhjp — APPROVED** [review]

**Comment by swenson** [comment]
Thanks!

**Pull request merged**